Templates/ScriptCanvasNode/Template/Source/${Name}.cpp

```diff
@@ -0,0 +1,23 @@
+// {BEGIN_LICENSE}
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+// {END_LICENSE}
+
+#include "${Name}.h"
+
+#include <ScriptCanvas/AutoGen/ScriptCanvasAutoGenRegistry.h>
+
+#include <Source/${Name}_Nodeables.generated.h>
+
+REGISTER_SCRIPTCANVAS_AUTOGEN_NODEABLE(${Name}Object);
```

Comment:
if I understand correctly, this file is for one nodeable class only, but `REGISTER_SCRIPTCANVAS_AUTOGEN_NODEABLE` is designed as self registry singleton object to be declared in system component or module class.

It is ok to put here, as we do have duplicate check, but not ideal. (as system will have multiple autogen registry objects but most of them are zombie object, as only one can be registered)

Reply:
It's convenient in that users don't need to manually modify the system component, but you do have a good point, it can be confusing as it's not required by the generated node itself. I could add a comment to explain it, or try to come up with an alternative that is also generated and doesn't require manual updates